apispec/insolvency-delta-spec.yml

```diff
@@ -0,0 +1,171 @@
+post:
+  summary: Accepts an incoming insolvency delta, transforms it into an avro schema and puts it onto a Kafka topic.
+  requestBody:
+    content:
+      application/json:
+        schema:
+          $ref: '#/components/schemas/Insolvencies'
+    required: true
+  responses:
+    "200":
+      description: Successfully produced message onto Kafka topic.
+    "400":
+      description: Bad request body - validation errors.
+    "401":
+      description: Unauthorised - missing api key in header.
+    "500":
+      description: Internal server error has occured.
+
+components:
+  schemas:
+    Insolvencies:
+      required: 
+      - insolvency
+      type: object
+      properties:
+        insolvency:
+          type: array
+          items:
+            $ref: '#/components/schemas/InsolvencyDelta'
+
+    InsolvencyDelta:
+      required:
+      - delta_at
+      - company_number
+      - case_numbers
+      type: object
+      properties:
+        delta_at:
+          type: string
+        company_number:
+          maxLength: 10
+          type: string
+        case_numbers:
+          type: array
+          items:
+            $ref: '#/components/schemas/CaseNumber'
+
+    CaseNumber:
+      required:
+      - case_number
+      - case_type_id
+      - case_type
+      type: object
+      properties:
+        case_number:
+          type: integer
+        case_type_id:
+          type: integer
+          enum:
+          - 1
+          - 2
+          - 3
+          - 5
+          - 6
+          - 7
+          - 8
+          - 13
+          - 14
+          - 15
+          - 17
+        case_type:
+          type: string
+          enum: ['Members Voluntary Liquidation',
+                'Creditors Voluntary Liquidation',
+                'Compulsory Liquidation',
+                'Receiver/Manager',
+                'Administrative Receiver',
+                'Administration',
+                'Corporate Voluntary Arrangement',
+                'In Administration',
+                'CVA Moratoria',
+                'Foreign Insolvency',
+                'Moratorium']
+        mortgage_id:
+          type: integer
```

Comment:
This is not large enough for actual data (defined as number(10)). Perhaps add format: int64 ?

Reply:
Done